### ISSUE TITLE
How to Stop Emulator?

### ISSUE THREAD

**Issue by cesarcristianodeoliveira** [emulatorjs/emulatorjs#105]
Hello I have your project on my react app, I navigate to pages without reload... so when changed the Emulator continues to play...
Can help me? My react with your project! Many Thanks! https://emulador.netlify.app/

One more question... how can put Portuguese Brazil?
Sorry for bad english!

Brazil here, thanks for all <3 God Bless U

**Comment by ethanaobrien** [comment]
Hello! I had a similar problem. Here is some code from [here](https://github.com/ethanaobrien/emulator-button/blob/main/decoded.js#L28) here that you can use. I changed a bit to support just calling this function to clear everything.

```
function decode {
    try {
        if (window.VARRRSSZZ) {
            var a = [];
            var b = [];
            for (var k in window) {
                a.push(k);
            }
            for (var i=0; i<a.length; i++) {
                if (! window.VARRRSSZZ.includes(a[i])) {
                    b.push(a[i]);
                }
            }
            for (var i=0; i<b.length; i++) {
                if (window[b[i]]) {
                    delete window[b[i]];
                }
            }
        }
        if (window.EJS_emulator && window.EJS_emulator.eventListeners) {
            for (var i=0; i<EJS_emulator.eventListeners.length; i++) {
                var a = EJS_emulator.eventListeners[i];
                a.element.removeEventListener(a.type, a.callback, a.capture);
            };
        };
    } catch(e) {};
    var a = [];
    for (var k in window) {
        a.push(k);
    }
    window.VARRRSSZZ = a;
}
```

call the decode function on page load (before emulator load) to properly init the variables it needs

>One more question... how can put Portuguese Brazil?

Are you looking to set the language of the UI?

**Comment by ethanaobrien** [comment]
@cesarcristianodeoliveira Do you think you could help translate all the localization variables (to whatever other language you can speak)? If you can just tell me and I'll tell you what to do

**Comment by cesarcristianodeoliveira** [comment]
Yes my friend, I'm from Brazil and I can translate to Spanish too
One more question, why I canno't run arcade games and the sound with PlayStation and N64 are not perfect?
Thanks for all again <3 God Bless U

**Comment by ethanaobrien** [comment]
@cesarcristianodeoliveira 
>One more question, why I canno't run arcade games

are you using the `arcade` or `mame` core? Just checking. 

>the sound with PlayStation and N64 are not perfect

Yes, this is a problem, the n64 isnt yet fixable. I am waiting for the updated n64 core to become avaliable. Sorry about this
I am working on an update that should fix the playstation audio.

>Yes my friend, I'm from Brazil and I can translate to Spanish too

Thank you.

So, could you download [this file](https://github.com/ethanaobrien/emulatorjs/blob/ethanaobrien-patch-1/data/localization/en.json) and rename it to whatever language you're translating to.
**Important**: Only translate the ones with the `-` at the start of the string. Once you translate all the text (the ones that start with the `-`) you can send it to me however you prefer.

Thank you for your help to make this more available!

**Comment by cesarcristianodeoliveira** [comment]
Here! Portuguese Brazil (pt-BR) and Spanish (es-ES)
[Portuguese Brazil and Espanol.zip](https://github.com/ethanaobrien/emulatorjs/files/7998196/Portuguese.Brazil.and.Espanol.zip)

I try with Arcade, with Mame and both not working

**Comment by ethanaobrien** [comment]
Thank you! 

What game are you trying to run?

**Comment by cesarcristianodeoliveira** [comment]
My code it's the same like this friend... But, my game don't play... I inclued the bios... but nothing :(
https://www.retrogames.onl/2018/01/cadillacs-and-dinosaurs-arcade.html
Sorry to to disturb

**Comment by ethanaobrien** [comment]
set `EJS_core` to `arcade`

Tell me if it works then

**Comment by cesarcristianodeoliveira** [comment]
Nothing with arcade
![image](https://user-images.githubusercontent.com/78006344/152446901-d6bee5a0-2a26-46c5-af5d-006ef44903e6.png)
![image](https://user-images.githubusercontent.com/78006344/152447004-138349cf-0121-4e46-84ee-7ad124ab2501.png)
And about the lang.json, you will make work on next update?

**Comment by ethanaobrien** [comment]
Yes, this will work on the next update. It looks like the rom might be bad

try the one from here https://play.retrogames.onl/arcade/games/dino.zip

**Comment by cesarcristianodeoliveira** [comment]
I tested dino.zip in your demo https://ethanaobrien.github.io/emulatorjs/ and not work too... I can't load any game with arcade or mame... sorry!
Keep with good work, thanks from Brazil.

**Comment by ethanaobrien** [comment]
Sorry, Im not sure what to do. Can you try re-downloading all the emulator files and if the problem still occurs could you send a screenshot of the network tab in the browser dev tools?

**Comment by cesarcristianodeoliveira** [comment]
Hy! Sorry to bother you...
I download again and I have the same console Error
If I emulated "dino.zip" in your demo, we have the same issue...
let's see in your demo, the same issue with Arcade Console and Dino.zip

Loading Rom in your demo
![image](https://user-images.githubusercontent.com/78006344/152691776-b3a08e79-be40-4786-9262-710c6669dee7.png)
And the Error is the same
![image](https://user-images.githubusercontent.com/78006344/152691800-09578563-1d1d-4b99-8788-37b3744ed9c0.png)

You're the best, don't forget!

**Comment by ethanaobrien** [comment]
Sorry, Im really not sure. #109 should resolve this. I'm going to mark as going to be resolved by #109

**Comment by ethanaobrien** [comment]
I added a `window.EJS_terminate` function
